docs/tutorial_basics.rst

```diff
@@ -0,0 +1,1087 @@
+
+
+.. code-block:: none
+
+   [ReFrame Setup]
+     version:           3.1-dev0 (rev: 986c3505)
+     command:           './bin/reframe -c tutorials/basics/hellomp/hellomp1.py -r'
+     launched by:       user@tresa.local
+     working directory: '/Users/user/Repositories/reframe'
+     settings file:     '/Users/user/Repositories/reframe/tutorials/config/settings.py'
+     check search path: '/Users/user/Repositories/reframe/tutorials/basics/hellomp/hellomp1.py'
+     stage directory:   '/Users/user/Repositories/reframe/stage'
+     output directory:  '/Users/user/Repositories/reframe/output'
+
+   [==========] Running 1 check(s)
+   [==========] Started on Mon Jun 22 00:58:27 2020
+
+   [----------] started processing HelloThreadedTest (HelloThreadedTest)
+   [ RUN      ] HelloThreadedTest on catalina:default using gnu
+   [ RUN      ] HelloThreadedTest on catalina:default using clang
+   [----------] finished processing HelloThreadedTest (HelloThreadedTest)
+
+   [----------] waiting for spawned checks to finish
+   [       OK ] (1/2) HelloThreadedTest on catalina:default using gnu [compile: 1.354s run: 1.250s total: 2.639s]
+   [       OK ] (2/2) HelloThreadedTest on catalina:default using clang [compile: 1.202s run: 0.238s total: 1.468s]
+   [----------] all spawned checks have finished
+
+   [  PASSED  ] Ran 2 test case(s) from 1 check(s) (0 failure(s))
+   [==========] Finished on Mon Jun 22 00:58:29 2020
+
+
+Everything looks fine, but let's inspect the actual output of one of the tests:
+
+
+.. code-block:: console
+
+   cat output/catalina/default/clang/HelloThreadedTest/rfm_HelloThreadedTest_job.out
+
+
+.. code-block:: none
+
+   [[[[    8] Hello, World!
+   1] Hello, World!
+   5[[0[ 7] Hello, World!
+   ] ] Hello, World!
+   [ Hello, World!
+   6[] Hello, World!
+   9] Hello, World!
+    2 ] Hello, World!
+   4] [[10 3] Hello, World!
+   ] Hello, World!
+   [Hello, World!
+   11] Hello, World!
+   [12] Hello, World!
+   [13] Hello, World!
+   [14] Hello, World!
+   [15] Hello, World!
+
+
+Not exactly what we were looking for!
+In the following we write a more robust sanity check that can catch this havoc.
+
+
+-----------------------------
+More advanced sanity checking
+-----------------------------
+
+Sanity checking of a test's outcome is quite powerful in ReFrame.
+So far, we have seen only a ``grep``-like search for a string in the output, but ReFrame's ``sanity_patterns`` are much more capable than this.
+In fact, you can practically do almost any operation in the output and process it as you would like before assessing the test's sanity.
+The syntax feels also quite natural since it is fully integrated in Python.
+
+In the following we extend the sanity checking of the multithreaded "Hello, World!", such that not only the output pattern we are looking for is more restrictive, but also we check that all the threads produce a greetings line.
```

Comment:
```suggestion
In the following we extend the sanity checking of the multithreaded "Hello, World!", such that not only the output pattern we are looking for is more restrictive, but we also check that all the threads produce a greetings line.
```

Reply:
Both are correct, except that mine puts a bit more emphasis in the second sentence. I will leave it as is.